contracts/src/LightWallet.sol

```diff
@@ -161,11 +167,32 @@ contract LightWallet is
         override
         returns (uint256 validationData)
     {
-        (bool isValid,) = _signatureValidation(userOpHash, userOp.signature);
-        if (!isValid) {
-            return SIG_VALIDATION_FAILED;
+        bytes1 signatureType = userOp.signature[0];
+
+        if (signatureType == 0x00 || signatureType == 0x01 || signatureType == 0x02 || signatureType == 0x03) {
```

Comment:
@fiveoutofnine

How to do this in bitwise op bro

Reply:
`if signatureType & 0x03 == signatureType`

Reply:
@shunkakinoki, thanks for the great question.

We can create a bitmap where `1` at the corresponding `signatureType` index denotes a valid type, and `0` does not:

```py
>>> hex(0b1111)
0xf
```

Thus, we have:

```sol
if ((0xf >> signatureType) & 1 == 1) {
    // Bruh
}
```

> [!NOTE]
> This also works for when `signatureType` is greater than `0x03` because all the bits in `0xf` get shifted off, and whatever you mask will equal to 0. Additionally, for future reference, you can often _directly use_ the value of `(0xf >> signatureType) & 1` to go into the next computation--very useful for removing branches.

Reply:
$MP

Reply:
Thank you so much, after much needed internal discussion, going w/ 

```
if signatureType & 0x03 == signatureType
```

thank you for your contributions